torch_xla/distributed/xla_dist.py

```diff
@@ -299,34 +314,56 @@ def _run_script(script_path, client_worker):
       thread.start()
       threads.append(thread)
 
-    try:
-      for thread in threads:
-        thread.join()
-    except KeyboardInterrupt:
-      pass
-    finally:
-      self._cleanup(script_map)
-
     for thread in threads:
       thread.join()
 
+  def _run_cmd(self, script_map):
+    try:
+      self._scp_scripts(script_map)
+      self._start_run(script_map)
+    except KeyboardInterrupt:
+      self.logger.warning(
+        'Child process received Ctrl^C. Exiting...',
+        extra={'clientip': '', 'ordinal': ''})
+      sys.exit(130)  # exit due to SIGINT from child proccess
```

Comment:
Maybe `128 + signal.SIGINT` might be clearer.

Reply:
Good idea. Done.